### ISSUE TITLE
Remote filter empty

### ISSUE THREAD

**Issue by DavidEstrela** [react-bootstrap-table/react-bootstrap-table2#370]
Hello,

I am using the event `onTableChange` to handle remote filters using query string, it works fine when a write something, the problem is when i delete some col filter text, i don't receive any filter event saying the affected row filter is `''` or `undefined`, something like that would be great, this would allow me to remove it from the query string

Regards and thanks in advance.

**Comment by AllenFang** [comment]
@davidestrela yes, that's right, because if the empty string was inputed, the `react-bootstrap-table` will drop/delete the column property on `filter` object instead of give it as empty/undefined.

Please do let me know your concern, every idea and suggestion is helpful to us. thanks 👍

**Comment by davidestrela** [comment]
Hello @AllenFang, as i said before i am using query string to store the table filters in order to rebuild the table in case of an hard reload for instance, the problem i am having is when i write more then one filter, the event `onTableChange` is fired one time with all the filters i have( what i whant ) and one time for each filter.

Example

| Filter1:'A' | Filter2:'B' |                  |
|--------- |-----------|---------- |
| A             | B               | B               |
| A             | A               | B               |

When i set the column filter `defaultValue` to the query string values(`Filter1=A`, `Filter2=B`), the event `onTableChange` will be fired three times,
1 - `filters` array will have [Filter1,Filter2] √
2 - `filters` array will have [Filter1]
3 - `filters` array will have [Filter2]

If the event `onTableChange` was fired only one time would solve my problem

As a workaround if every time an input filter text is deleted could be fired the `onTableChange` with the value set to `undefined` or `''` and the behaviour is done by the app

Thanks

**Comment by AllenFang** [comment]
@davidestrela understand, i will spend some times to evaluate this situation. BTW, if in this case(remote mode + default value filter), is it better to you if `onTableChange` never called when you give default value but you need to pre filter data? because for remote mode, I think sometime it's make sense to prefilter data by yourself if you have default value so that its able to avoid to call `onTableChange`.
